sdp/processors/datasets/masc/aggregate_segments.py

```diff
@@ -0,0 +1,123 @@
+    
+    output_audio_dir (str): Directory where aggregated audio segments will be stored.
+        If `crop_audio_segments` is False, this path is used to create the audio file paths in the manifest.
+    input_segments_key (str): The field name that contains list of segments in the input manifest. Defaults to "segments".
+    input_audio_filepath_key (str): The field name that contains paths to the audio files in the input manifest.
+        Defaults to "audio_filepath".
+    output_text_key (str): Field name where to store aggregated segment text. Defaults to "text".
+    output_duration_key (str): Field name where aggregated segment durations will be stored. Defaults to "duration".
+    output_audio_filepath_key (str): Field name where aggregated segment audio file paths will be stored.
+        Defaults to "audio_filepath".
+    max_duration (float): Maximum duration of aggregated segment. Default to 20.0s.
+    crop_audio_segments (bool): Flag indicating whether to crop audio files according to the aggregated segments.
+        Defaults to True.
+    verbose (bool): Set to True to enable more detailed logging. Defaults to False.
+    """
+    def __init__(
+        self,
+        output_audio_dir: str,
+        input_segments_key: str = "segments",
+        input_audio_filepath_key: str = "audio_filepath",
+        output_text_key: str = "text",
+        output_duration_key: str = "duration",
+        output_audio_filepath_key: str = "audio_filepath",
+        max_duration: float = 20.0,
+        crop_audio_segments: bool = True,
+        verbose: bool = False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.max_duration = max_duration
+        self.input_audio_filepath_key = input_audio_filepath_key
+        self.output_splitted_audio_filepath_key = output_audio_filepath_key
+        self.crop_audio_segments = crop_audio_segments
+        self.output_audio_dir = output_audio_dir
+        self.input_segments_key = input_segments_key
+        self.verbose = verbose
+        self.output_text_key = output_text_key
+        self.output_duration_key = output_duration_key
+
+    def prepare(self):
+        if self.crop_audio_segments and self.output_audio_dir:
+            os.makedirs(os.path.join(self.output_audio_dir), exist_ok=True)
+
+    def process_dataset_entry(self, data_entry: dict):
+        if self.input_segments_key not in data_entry:
+            if self.verbose:
+                logging.info(f"No segements in a sample {data_entry[self.input_audio_filepath_key]}.")
+            return []
+
+        segments = data_entry[self.input_segments_key]
+        if len(segments) == 0:
+            return []
+        
+        audio_basename = os.path.basename(data_entry[self.input_audio_filepath_key]).split(".")[0]
+        agg_segments = []
+        aggregated_segment = {**segments[0]}
+        for segment in segments[1:]:
+            # checking if after adding segement it's duration will exceed `max_duration`
```

Comment:
segment

Reply:
maybe:
"# checking if adding another segment will cause the total duration to exceed `max_duration`"